src/librustc_ast_lowering/expr.rs

```diff
@@ -521,7 +521,7 @@ impl<'hir> LoweringContext<'_, 'hir> {
 
     /// Desugar `<expr>.await` into:
     /// ```rust
-    /// match <expr> {
+    /// match ::std::future::IntoFuture::into_future(<expr>) {
```

Comment:
I don't see how #67757 is possible, the only HIR difference here should be handled by #65743.